src/server/components/dashboard/annualReview/helpers.ts

```diff
@@ -22,57 +22,59 @@ function addSixMonths(date: number | string | Date): Date {
   return newDate;
 }
 
-export function getAnnualReviewYear({
-  day,
-  month,
-  lastAnnualReview,
+export function calculateValidDate({
+  userEnteredDate,
+  maxDate,
 }: {
-  day?: string;
-  month?: string;
-  lastAnnualReview: number;
-}): number {
-  const date = new Date(lastAnnualReview);
+  userEnteredDate: Date;
+  maxDate: Date;
+}): Date | null {
+  const userEnteredDateNextYear = DateFns.add(userEnteredDate, { years: 1 });
```

Comment:
not sure I understand the point of this.  If you add 1 year to the userEnteredDate then it will never be returned from possibleDates.filter on line 35 so what's the point of it?

Reply:
OK so you add two dates to the possibleDates array, one for the current year, one for the next year and expect only one of them to return.  Easier than calculating the current month to conditionally determine whether to increment. Makes sense.